cEP-0024.md

```diff
@@ -0,0 +1,129 @@
+to any of the issues.
+
+### Auto-reject PR opened on master of forks
+
+Normally, contributors should make changes in different branch from master
+branch of the forked repo, and submit a PR. But, sometimes coala receives PRs
+from master branch of the forked repo.
+
+coala needs mechanism by which GitMate automatically closes PRs opened on
+master branch of forked repo, marks certain label and comments on the PR
+informing the PR author to work on a different branch. Comment message, label
+will be configurable.
+
+### Auto-mark PR with necessary labels as per the reviews done
+
+When a reviewer reviews a PR, he/she either approves changes or request changes
+and leaves it there, but the PR would be still in the review queue. So, coala
+needs better mechanism to clear review queue when anyone reviews PR or when
+there is any CI failure. GitMate will automatically mark labels as per the
+following conditions:
+
+- WIP label conditions:
+  1. Reviewer requests changes
+  2. GitMate's coala plugin finds defects
+  3. CI tests fail
+- Approve label conditions:
+  1. Reviewer approves changes
+
+### Fixes vs Closes automated checking
+
+Newcomers often confuse when they should use `Fixes` or `Closes` in a commit
+message. So, GitMate should automatically check every commit for issues which
+are being `Fixed` or `Closed`, if those issues have a label (configurable)
+which represents bug, `Fixes` should be used, else `Closes` should be used.
+
+When the commit message contains 'Fixes', but referenced issue doesn't have
+label(configurable) which represents bug, then a comment(configurable) will be
+added on the PR. Message to be commented can be configured by coala as:
+
+```
+`Fixes` is used in commmit message, but referenced issue doesn't have a
+`type/bug` label, if issue is updated to include the `type/bug` label, then
+ask a maintainer to add required label, else use `Closes`.
+```
+
+When the commit message contains 'Closes', but referenced issue has bug label,
+then a comment(configurable) will added on the PR. Message to be commented
+can be configured by coala as:
+
+```
+`Closes` is used, but referenced issue has a `type/bug` label, if issue is
+updated to remove `type/bug` label, then ask a maintainer to remove required
+label, else use `Fixes`.
+```
+
+In both cases, 'process/pending review' label will be removed and 'process/wip'
+label will be added. This will be part of "Assign label as per review state"
+plugin.
+
+### Assign Plugin
+
+When a user comments on an issue requesting for assignment to the issue,
+GitMate will assign comment author to the issue, if he/she is eligible to get
+assigned, else missing eligibility conditions will be specified as a comment.
+Will support configurations for limiting number of certain labelled issues a
+member can work on, block assignment for certain users or certain labelled
+issues, difficulty level order to be followed while assigning issues to members
```

Comment:
huge paragraphs are scary, and people just skim them.
split this into paragraphs.

help the reader understand what is being written by letting them finish one small bit and pause, and then attempt to read the next chunk.

splitting it also helps the review process as one paragraph can become stable, while the others  are still being revised.